rlberry/envs/classic_control/SpringCartPole.py

```diff
@@ -0,0 +1,590 @@
+"""
+SpringCartPole environment introduced in J-F. Hren PhD thesis.
+"""
+
+import numpy as np
+import rlberry.spaces as spaces
+from rlberry.envs.interface import Model
+from rlberry.rendering import Scene, GeometricPrimitive, RenderInterface2D
+from rlberry.rendering.common_shapes import bar_shape, circle_shape
+
+
+class SpringCartPole(RenderInterface2D, Model):
+    """
+    SpringCartPole is an extension of the CartPole environment proposed in
+    PhD thesis J-F. Hren. It consists of two carts connected by a spring.
+
+    Parameters
+    ----------
+    dt : float, defualt=0.02
+        Time step of the simulation.
+    obs_trans : bool, default=True
+        If True, state has dimension 10:
+        State:
+            'Cart position1', 'Cart velocity1', 'Pole cos1', 'Pole sin1', 'Pole angular velocity1',
+            'Cart position2', 'Cart velocity2', 'Pole cos2', 'Pole sin2', 'Pole angular velocity2'
+        If False, state has dimension 8:
+        State:
+            'Cart position1', 'Cart velocity1', 'Pole angle1', 'Pole angular velocity1',
+            'Cart position2', 'Cart velocity2', 'Pole angle2', 'Pole angular velocity2'
+    swing_up : bool, default=False
+        If True, the pole starting position is at the bottom
+        If False, the pole starting position is at the top
+    random_init : bool, default=True
+        If True, the noise is added to the carts and poles starting positions
+
+    Notes
+    -----
+    State:
+        The state consists of the position of cart 1, its speed, the angle
+        of pole 1 (expressed in radians or in a tuple of cos() and sin()) and
+        its angular speed, and the same set of values for cart 2 and pole 2.
+        For both poles, the angle of 0 corresponds to the vertical position,
+        the positive angles correspond to a counterclockwise rotation.
+
+    Actions:
+        The action is either 0, 1, 2, or 3, corresponding to the four possible
+        actions:
+            LL = 0, move cart 1 to the left, cart 2 to the left
+            RR = 1, move cart 1 to the right, cart 2 to the right
+            LR = 2, move cart 1 to the left, cart 2 to the right
+            RL = 3, move cart 1 to the right, cart 2 to the left
+        The magnitude of actions is fixed to 2.0.
+
```

Comment:
Maybe a note on how the reward is computed ? I imagine it is similar to cartpole but it is not clear because there are two poles.